include/ipc/ipc_service.h

```diff
@@ -170,6 +170,80 @@ int ipc_service_register_endpoint(const struct device *instance,
  */
 int ipc_service_send(struct ipc_ept *ept, const void *data, size_t len);
 
+/** @brief Get an empty TX buffer to be sent using @ref ipc_service_send_nocopy
+ *
+ *  This function can be called to get an empty TX buffer so that the
+ *  application can directly put its data into the sending buffer without copy
+ *  from an application buffer.
+ *
+ *  It is the application responsibility to correctly fill the allocated TX
+ *  buffer with data and passing correct parameters to @ref
+ *  ipc_service_send_nocopy function to perform data no-copy-send mechanism.
+ *
+ *  @param ept Registered endpoint by @ref ipc_service_register_endpoint.
+ *  @param data Pointer to the empty TX buffer.
+ *  @param len Pointer to store the TX buffer size.
```

Comment:
Shouldn't it be an input where you provide requested length? Or [in,out] where on output granted length (equal or bigger than requested) is stored.

Reply:
but #43544 is using circular buffer so you need to know how much space to allocate because there is a variable length allocation.

Reply:
I think we need only one parameter (`reqested`) one (for me its okay if its named `len`), and this function should return `-ENOBUFS` or `-ENOMEM` if requested buffer can not be claimed withing `k_timeout_t timeout`. I think there is no point to add `returned` size, because user needs specific size, if cannot be provided error must be returned.